### PR TITLE
Test start script

### PR DESCRIPTION
**Pull request by Profpatsch**
First commit is only a refactor. Second one changes start script to have a working example haskell binary, last one adds a test.

### CODE REVIEW

File: tests/start-script/BUILD [hunk 1]
```diff
@@ -0,0 +1,9 @@
+load(
+    "//:bazel/inline_tests.bzl",
+    "sh_inline_test"
```
Comment (NinjaTrappeur) [inline tests/start-script/BUILD]: Do we really need `sh_inline_test` in scope here?

Comment (Profpatsch) [reply tests/start-script/BUILD]: We don’t, if you can find another way to make the start script build a Haskell binary without writing the source file to the user’s repository

**Comment by mboes** [comment]
A lot of extra test weight here, including putting litter in the user's repository. Can you state the problem you're trying to fix?

**Comment by Profpatsch** [comment]
> including putting litter in the user's repository

Huh, where?

**Comment by mboes** [comment]
> Huh, where?

My bad - misintepretation of a comment your wrote above.

> The `start` script broke in the past, because the `rules_haskell` API changed.

We already test the fetch [here](https://github.com/tweag/rules_haskell/blob/master/tests/RunTests.hs#L52-L69). Can't we extend that to test the build?

**Comment by Profpatsch** [comment]
> We already test the fetch [here](https://github.com/tweag/rules_haskell/blob/master/tests/RunTests.hs#L52-L69). Can't we extend that to test the build?

Unless the fetch tests some different state, this tests is a complete superset of that test, running inside of the `bazel test //...` instead of a different bazel run.

**Comment by mboes** [comment]
And also adding extra test related code that we can do without.

**Comment by Profpatsch** [comment]
So after talking to @mboes, the way we go forward with this is the following: Bazel sets the current repository boundary to where it finds another `WORKSPACE` file. This means you can use `local_repository` to import a subdirectory (`c2hs` already does that).

The plan forward:
1. Change the start script to be an a workspace directory.
1. test this workspace
1. Generate the `start` script from these files

We can use the same strategy to migrate https://github.com/tweag/rules_haskell_examples back into the `rules_haskell` repository.

**Pull request closed without merging**